eip-0023/eip-0023.md

```diff
@@ -0,0 +1,337 @@
+      - Registers R0 (value), R1 (script), and the first token (refresh NFT) are preserved from the old refresh box.
+3. Each input oracle box has following logic ([smart contract](#oracle-contract)):
+   - The first input is a pool box (i.e., has the pool token).
+   - An output oracle box (acting as a copy of this box) must be created as follows:
+      - The following values are directly copied: R0 (nanoErgs), R1 (script), the first token (oracle token) (stored in R2) and R4 (GroupElement).
+      - The second token (reward token) is incremented by at least 1.
+
+Suppose there are *n* valid oracle boxes, then there are 2*n* reward tokens released.   
+It is expected that whoever creates the refresh transaction (the *collector*) takes *n*+1 reward tokens, and the other oracles get 1 token each.  
+This gives incentive to use as many oracle boxes as possible during the refresh.
+
+### Publish data-point
+
+| Index | Input | Output |
+|---|---|---|
+| 0 | Oracle | Oracle |  
+
+This allows an oracle to publish data-point for collection in next epoch.
+This entails spending the oracle box and creating a new oracle box as per the [smart contract](#oracle-contract).
+
+1. The public key stored in R4 defines who can spend the oracle box.
+2. The logic requires the new oracle box to be as follows:
+   - The script and the first token are copied from this box.
+   - R4 contains a group element.
+   - The second token is the reward token in some non-zero quantity.
+3. The following rules (not enforced by the smart contract) to be followed.
+   - It should store the same group element in R4.
+   - It should store epoch counter of the current pool box in R5.
+   - It should store the rate to publish in R6.
+   - It should ensure reward tokens are preserved.
+
+### Extract reward tokens
+
+| Index | Input | Output |
+|---|---|---|
+| 0 | Oracle | Oracle |
+| 1 |             | Box with freed reward tokens |
+
+This allows an oracle to extract tokens obtained from publishing data point to redeem them elsewhere.
+
+The transaction is similar to the [publish data-point](#publish-data-point) transaction, except that Step 3 is modified as follows.
+
+3. The following rules (not enforced by the smart contract) to be followed.
+   - It should store the same group element in R4.
+   - It should keep at least 1 reward token.
+   - It should store the balance reward tokens in some other box.
+
+
+### Transfer oracle token
+
+| Index | Input | Output |
+|---|---|---|
+| 0 | Oracle | Oracle |  
+
+This is used to transfer the ownership of an oracle token to another public key.
+
+The transaction is similar to the [publish data-point](#publish-data-point) transaction, except that Step 3 is modified as follows.
+
+3. The following rules (not enforced by the smart contract) to be followed.
+   - It should store the new owner's group element in R4.
+   - It should keep at least 1 reward token.
+   - It should store the balance reward tokens, if any, in some other box.
```

Comment:
Following our discussion in https://github.com/ergoplatform/eips/pull/41#discussion_r870469418
```suggestion
   - It should check that there is only 1 reward token is in the oracle box and send it to the new owner along with oracle token.
```